catalog/plugin/catalog-plugin-expirationdate/src/test/java/org/codice/ddf/catalog/plugin/expiration/TestExpirationDatePlugin.java

```diff
@@ -0,0 +1,597 @@
+        assertThat(mockCreateRequest.getMetacards().size(), is(numberOfMetacards));
+        assertNotNull(mockCreateRequest.getMetacards().get(0));
+        DateTime newExpirationDate = new DateTime(
+                mockCreateRequest.getMetacards().get(0).getExpirationDate());
+        // New exp date is 10 days after modified date
+        assertTrue(newExpirationDate.equals(MODIFIED_DATE.plusDays(days)));
+    }
+
+    @Test
+    /**
+     * Tests the process(CreateRequest createRequest) method.
+     *
+     * Overwrite the 'blank' expiration date (single metacard).  The new expiration date is 10 after the modified date.
+     *
+     * offsetFromModifiedDate: 10
+     * overwriteIfBlank: true
+     * overwriteIfExists: true
+     *
+     * @throws PluginExecutionException
+     * @throws StopProcessingException
+     */ public void testProcessCreateRequestMetacardHasNoExpirationDateOverwriteIfBlankIsTrueOverwriteIfExistsIsTrueDaysOffsetFromModifiedDateIs10()
+            throws PluginExecutionException, StopProcessingException {
+
+        // Setup
+        ExpirationDatePlugin expirationDatePlugin = new ExpirationDatePlugin();
+
+        // Items configured via the admin console
+        int days = 10;
+        expirationDatePlugin.setOffsetFromModifedDate(days);
+        expirationDatePlugin.setOverwriteIfBlank(true);
+        expirationDatePlugin.setOverwriteIfExists(true);
+
+        CreateRequest mockCreateRequest = mock(CreateRequest.class);
+        int numberOfMetacards = 1;
+        List<Metacard> mockMetacards = createMockMetacardsWithNoExpirationDate(numberOfMetacards);
+        when(mockCreateRequest.getMetacards()).thenReturn(mockMetacards);
+
+        // Perform Test
+        expirationDatePlugin.process(mockCreateRequest);
+
+        // Verify
+        assertThat(mockCreateRequest.getMetacards().size(), is(numberOfMetacards));
+        assertNotNull(mockCreateRequest.getMetacards().get(0));
+        DateTime newExpirationDate = new DateTime(
+                mockCreateRequest.getMetacards().get(0).getExpirationDate());
+        // New exp date is 10 days after modified date
+        assertTrue(newExpirationDate.equals(MODIFIED_DATE.plusDays(days)));
+    }
+
+    /**
+     * Tests the process(CreateRequest createRequest) method.
+     * <p/>
+     * Don't overwrite the 'blank' expiration dates (multiple metacards) since the overwriteIfBlank option is not selected.
+     * <p/>
+     * offsetFromModifiedDate: 10
+     * overwriteIfBlank: false
+     * overwriteIfExists: false
+     *
+     * @throws PluginExecutionException
+     * @throws StopProcessingException
+     */
+    @Test
+    public void testProcessCreateRequestMetacardHasNoExpirationDateOverwriteIfBlankIsFalseOverwriteIfExistsIsFalseDaysOffsetFromModifiedDateIs10MultipleMetacards()
+            throws PluginExecutionException, StopProcessingException {
+
+        // Setup
+        ExpirationDatePlugin expirationDatePlugin = new ExpirationDatePlugin();
```

Comment:
All of these setups can be consolidated into an `@Before` method (which is called before each test)